pex/pip/version.py

```diff
@@ -65,16 +85,51 @@ def requires_python_applies(self, target):
 class LatestPipVersion(object):
     def __get__(self, obj, objtype=None):
         if not hasattr(self, "_latest"):
-            self._latest = max(PipVersionValue._iter_values(), key=lambda pv: pv.version)
+            self._latest = max(
+                (version for version in PipVersionValue._iter_values() if not version.hidden),
+                key=lambda pv: pv.version,
+            )
         return self._latest
 
 
+class DefaultPipVersion(object):
```

Comment:
Previously VENDORED was the global default but that no longer works. This allows dynamically choosing the correct default given the current ambient interpreter: VENDORED for everything <3.12, unreleased Pip (for now) for everything newer.